### PR TITLE
#64306 pass event data from director to reader

### PR DESCRIPTION
**Pull request by tabman83**
Change EventReaderInitData and pass domain configuration from Director to Reader.

### CODE REVIEW

**Comment by tabman83** [comment]
It suffers of the same issue we spotted with Bullfrog so had to downgrade Microsoft.Extensions.Configuration.AzureKeyVault to v.2.2.0

File: src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs [hunk 1]
```diff
@@ -9,38 +12,59 @@ public class EventReaderInitData
...
+                EventType = subscriberConfiguration.EventType,
+                DlqMode = subscriberConfiguration.DLQMode,
+                SourceSubscription = subscriberConfiguration.DLQMode != null ? subscriberConfiguration.SourceSubscriptionName : null
+            };
+        }
+
+        public byte[] ToByteArray()
+        {
+            var settings = new JsonSerializerSettings();
+            settings.ContractResolver = new JsonIgnoreAttributeIgnorerContractResolver();
```
Comment (mszymeshop) [inline src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs:45]: Looks like the resolver does not change in the lifetime of the class and can be made static

File: src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs [hunk 2]
```diff
@@ -9,38 +12,59 @@ public class EventReaderInitData
...
+            var stringValue = JsonConvert.SerializeObject(this, settings);
+            var byteArray = Encoding.UTF8.GetBytes(stringValue);
+            return byteArray;
+        }
+
+        public static EventReaderInitData FromByteArray(byte[] buffer)
+        {
+            var content = Encoding.UTF8.GetString(buffer);
+            var settings = new JsonSerializerSettings();
+            settings.ContractResolver = new JsonIgnoreAttributeIgnorerContractResolver();
```
Comment (mszymeshop) [inline src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs:55]: as above

File: src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs [hunk 3]
```diff
@@ -9,38 +12,59 @@ public class EventReaderInitData
...
+            settings.Converters.Add(new AuthenticationConfigConverter());
```
Comment (mszymeshop) [inline src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs:56]: Looks like the converter does not change through the lifetime of the class and can be made static

File: src/CaptainHook.DirectorService/DirectorService.cs [hunk 4]
```diff
@@ -77,13 +77,17 @@ protected override async Task RunAsync(CancellationToken cancellationToken)
...
-                foreach (var (key, sub) in _subscriberConfigurations)
+                foreach (var (key, subscriber) in _subscriberConfigurations)
                 {
                     if (cancellationToken.IsCancellationRequested) return;
 
-                    var readerServiceNameUri = ServiceNaming.EventReaderServiceFullUri(sub.EventType, sub.SubscriberName, sub.DLQMode != null);
+                    var readerServiceNameUri = ServiceNaming.EventReaderServiceFullUri(subscriber.EventType, subscriber.SubscriberName, subscriber.DLQMode != null);
                     if (!serviceList.Contains(readerServiceNameUri))
                     {
+                        var initializationData = EventReaderInitData
```
Comment (mszymeshop) [inline src/CaptainHook.DirectorService/DirectorService.cs:87]: :clap:

File: src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs (hunk 1 above)
Comment (tabman83) [reply src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs]: I have changed that.

File: src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs (hunk 3 above)
Comment (tabman83) [reply src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs]: I have changed that.

File: src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs (hunk 2 above)
Comment (tabman83) [reply src/CaptainHook.Common/ServiceModels/EventReaderInitData.cs]: I have changed that.

**Pull request merged**